cpp/src/slice2cs/Gen.cpp

```diff
@@ -3051,9 +3130,9 @@ Slice::Gen::ClassFactoryVisitor::visitClassDefStart(const ClassDefPtr& p)
     string name = fixId(p->name());
     _out << sp;
     emitCommonAttributes();
-    _out << nl << "public class " << name << " : ZeroC.Ice.IClassFactory";
+    _out << nl << "public class " << name << " : global::ZeroC.Ice.IClassFactory";
```

Comment:
I added these `global::` because the new optional test failed to compile without. It's only/primarily an issue for the class and remote exception factories generated in `ZeroC.Ice.ClassFactory` resp. `ZeroC.Ice.RemoteExceptionFactory`. Here the full namespace is `ZeroC.Ice.ClassFactory.<mapped module name>` i.e. `ZeroC.Ice.ClassFactory.ZeroC.Ice.Test.Optional` and within this namespace, a lookup to `ZeroC.Ice.` searches the nested `ZeroC.Ice`.